File: prisma/refs/search-a-2d-matrix.ts
export function searchMatrix(matrix: number[][], target: number): boolean {
  const m = matrix.length, n = matrix[0]?.length ?? 0;
  let lo = 0, hi = m * n - 1;
  while (lo <= hi) {
    const mid = (lo + hi) >> 1;
    const v = matrix[(mid / n) | 0][mid % n];
    if (v === target) return true;
    if (v < target) lo = mid + 1;
    else hi = mid - 1;
  }
  return false;
}

File: prisma/refs/py/search-a-2d-matrix.py
def search_matrix(matrix, target):
    m, n = len(matrix), len(matrix[0]) if matrix else 0
    lo, hi = 0, m * n - 1
    while lo <= hi:
        mid = (lo + hi) // 2
        v = matrix[mid // n][mid % n]
        if v == target:
            return True
        if v < target:
            lo = mid + 1
        else:
            hi = mid - 1
    return False
